lib/services/api/api_service.ts

```diff
@@ -314,4 +316,41 @@ export class SNApiService extends PureService {
     return result;
   }
 
+  async getItemRevisions(itemId: string) {
+    if (!this.session) {
+      return undefined;
```

Comment:
Shouldn't we return an error response, like we do in sync and all other API requests? Also, what if `this.refreshingSession` is true here? Maybe we should make a general function for basic checks, like this:

```javascript
preprocessingError() {
  if (this.refreshingSession) {
    return this.createErrorResponse(messages.API_MESSAGE_TOKEN_REFRESH_IN_PROGRESS);
  }
  if (!this.session) {... }

  return undefined;
}
```

Then in all API requests, you do:

```javascript
const preprocessingError = this.preprocessingError();
if (preprocessingError) { return preprocessingError }
```